app/src/main/java/berlin/mfn/naturblick/ui/fieldbook/fieldbook/FieldbookMapFragment.kt

```diff
@@ -0,0 +1,212 @@
+import com.mapbox.maps.ViewAnnotationAnchor
+import com.mapbox.maps.extension.style.layers.properties.generated.IconAnchor
+import com.mapbox.maps.plugin.annotation.annotations
+import com.mapbox.maps.plugin.annotation.generated.PointAnnotationManager
+import com.mapbox.maps.plugin.annotation.generated.PointAnnotationOptions
+import com.mapbox.maps.plugin.annotation.generated.createPointAnnotationManager
+import com.mapbox.maps.plugin.gestures.OnMoveListener
+import com.mapbox.maps.plugin.gestures.addOnMapClickListener
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.plugin.viewport.data.FollowPuckViewportStateOptions
+import com.mapbox.maps.plugin.viewport.viewport
+import com.mapbox.maps.viewannotation.viewAnnotationOptions
+import kotlinx.coroutines.flow.collectLatest
+import java.util.*
+import kotlin.math.roundToInt
+import kotlinx.coroutines.launch
+
+class FieldbookMapFragment : Fragment() {
+    private lateinit var binding: FragmentFieldbookMapBinding
+    private lateinit var viewModel: FieldbookViewModel
+    private lateinit var mapView: MapView
+    private lateinit var pointAnnotationManager: PointAnnotationManager
+    private val onMoveListener: OnMoveListener = object : OnMoveListener {
+        override fun onMove(detector: MoveGestureDetector): Boolean {
+            return false
+        }
+
+        override fun onMoveBegin(detector: MoveGestureDetector) {
+            viewModel.stopTracking()
+        }
+
+        override fun onMoveEnd(detector: MoveGestureDetector) {}
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val initialOccurenceId = IntentCompat.getParcelableExtra(
+            requireActivity().intent,
+            OCCURENCE_ID,
+            ParcelUuid::class.java
+        )?.uuid
+        val model: FieldbookViewModel by activityViewModels {
+            FieldbookViewModelFactory(initialOccurenceId, requireActivity().application)
+        }
+        viewModel = model
+        val popupOffset = resources.getDimension(R.dimen.popup_offset).roundToInt()
+        lifecycleScope.launch {
+            viewModel.observationAndSelected.flowWithLifecycle(lifecycle)
+                .collectLatest { (observations, selected) ->
+                    val mapView = binding.mapView
+                    val map = mapView.getMapboxMap()
+                    map.addOnMapClickListener {
+                        mapView.viewAnnotationManager.removeAllViewAnnotations()
+                        false
+                    }
+                    pointAnnotationManager.deleteAll()
+                    mapView.viewAnnotationManager.removeAllViewAnnotations()
+                    val annotations: List<PointAnnotationOptions> =
+                        observations.mapNotNull { observation ->
+                            observation.coords?.let { coords ->
+                                observation.species?.group?.let { group ->
```

Comment:
if no species is set you should set the unknown `?` marker.

Reply:
Good catch!